[Frontend] Adding the "User Defined Custom Tool Calling" parser for the Llama models

lulmer opened the pull request:
## Description

The current Llama tool parsing in vLLM is based on the [JSON based tool calling](https://llama.meta.com/docs/model-cards-and-prompt-formats/llama3_1/#json-based-tool-calling) using the procedure given by Meta. However, another tool parsing strategy is mentioned on this same website : The [User Defined Custom Tool Calling](https://www.llama.com/docs/model-cards-and-prompt-formats/llama3_1/#user-defined-custom-tool-calling). 

The gain is substantial : After testing this approach as a plugin on a private function calling benchmark (more than 120 different scenarios tested with a set of 30 complex and lengthy fintech tool definitions), I observed significantly higher function-calling accuracy compared to the current JSON-based tool parser. I also run some experiments on the BFCL benchmark (AST non-live bench) and could observe the same types of improvements:
 
| Model                                          | AST Summary | Simple AST | Python Simple AST | Java Simple AST | JavaScript Simple AST | Multiple AST | Parallel AST | Parallel Multiple AST |
| ---------------------------------------------- | ----------- | ---------- | ----------------- | --------------- | --------------------- | ------------ | ------------ | --------------------- |
| Meta-Llama-3.1-8B-Instruct-FC (json tool call) | 43.33%      | 56.33%     | 39.00%            | 60.00%          | 70.00%                | 29.50%       | 30.00%       | 18.50%                |
| Meta-Llama-3.1-8B-Instruct-(usr def custom tc)           | 72.98%      | 67.58%     | 89.75%            | 57.00%          | 56.00%                | 86.50%       | 78.50%       | 75.50%                |

This PR introduces a new `Llama3UserDefinedCustomToolParser` class that extends the `ToolParser` base class. The new parser allows for streaming support when using custom tools with the Llama models. It handles the extraction of tool calls and arguments from the model's response in streaming too, enabling real-time processing of tool calls.

The flow looks like this : 
![image](https://github.com/user-attachments/assets/5ee5965a-afe7-4dc6-a1ac-438859ad4832)


## Main Changes

1. **New Parser Class**: The `Llama3UserDefinedCustomToolParser` class is added to handle streaming tool calls for Llama models.
2. **New Chat Template**: An example of chat template that works well with this parser can be found here : `vllm/examples/tool_chat_template_llama3.1_usr_def_tool_call.jinja`


## Remarks

This is my first PR on the vLLM project and I believe there is still some stuff I need guidance on : 
- What is the testing process ? I don't really know how to proceed
- This parser requires a chat template that is different from the one that has been written for the JSON based tool calling although I provided it in the example directory I don't know how to indicate to the user how to use it. 
- There might be some edge case I am not covering when handling the streaming tool parsing. 
- I noticed there is a refactoring issue for the frontend going on https://github.com/vllm-project/vllm/issues/11522 and discussion over potential benefits of using FSM for tool parsers, I don't know how it could impact the work here but I am happy to discuss potential changes.

njhill committed:
[V1][Sampler] Faster top-k only implementation (#15478)

Signed-off-by: Nick Hill <nhill@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

dr75 committed:
Support SHA256 as hash function in prefix caching (#15297)

Signed-off-by: Marko Rosenmueller <5467316+dr75@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Alexei-V-Ivanov-AMD committed:
Applying some fixes for K8s agents in CI (#15493)

Signed-off-by: Alexei V. Ivanov <alexei.ivanov@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexm-redhat committed:
[V1] TPU - Revert to exponential padding by default (#15565)

Signed-off-by: Alexander Matveev <amatveev@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexm-redhat committed:
[V1] TPU CI - Fix test_compilation.py (#15570)

Signed-off-by: Alexander Matveev <amatveev@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wrmedford committed:
Use Cache Hinting for fused_moe kernel (#15511)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yaochengji committed:
[TPU] support disabling xla compilation cache (#15567)

Signed-off-by: Chengji Yao <chengjiyao@google.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

MattTheCuber committed:
Support FIPS enabled machines with MD5 hashing (#15299)

Signed-off-by: Matthew Vine <32849887+MattTheCuber@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ElizaWszola committed:
[Kernel] CUTLASS grouped gemm fp8 MoE kernel (#13972)

Signed-off-by: ElizaWszola <eliza@neuralmagic.com>
Signed-off-by: ElizaWszola <ewszola@redhat.com>
Co-authored-by: Lucas Wilkinson <wilkinson.lucas@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
Add automatic tpu label to mergify.yml (#15560)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Chenyaaang committed:
add platform check back (#15578)

Signed-off-by: Chenyaaang <llccyy1212@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

varun-sundar-rabindranath committed:
[misc] LoRA: Remove unused long context test data (#15558)

Signed-off-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Co-authored-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wayzeng committed:
[Doc] Update V1 user guide for fp8 kv cache support (#15585)

Signed-off-by: weizeng <weizeng@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

MengqingCao committed:
[moe][quant] add weight name case for offset (#15515)

Signed-off-by: Mengqing Cao <cmq0113@163.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

comaniac committed:
[V1] Refactor num_computed_tokens logic (#15307)

Signed-off-by: Cody Yu <hao.yu.cody@gmail.com>
Co-authored-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jerryzh168 committed:
Allow torchao quantization in SiglipMLP (#15575)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm] Env variable to trigger custom PA (#15557)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yaochengji committed:
[TPU] [V1] fix cases when max_num_reqs is set smaller than MIN_NUM_SEQS (#15583)

Signed-off-by: Chengji Yao <chengjiyao@google.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ruisearch42 committed:
[Misc] Restrict ray version dependency and update PP feature warning in V1 (#15556)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[TPU] Avoid Triton Import (#15589)

Signed-off-by: rshaw@neuralmagic.com <robertgshaw2@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Avabowler committed:
[Misc] Consolidate LRUCache implementations (#15481)

Signed-off-by: Bella kira <2374035698@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[Quantization] Fp8 Channelwise Dynamic Per Token GroupedGEMM (#15587)

Signed-off-by: ElizaWszola <eliza@neuralmagic.com>
Signed-off-by: ElizaWszola <ewszola@redhat.com>
Signed-off-by: rshaw@neuralmagic.com <robertgshaw2@gmail.com>
Co-authored-by: ElizaWszola <eliza@neuralmagic.com>
Co-authored-by: Lucas Wilkinson <wilkinson.lucas@gmail.com>
Co-authored-by: ElizaWszola <ewszola@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Misc] Clean up `scatter_patch_features` (#15559)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

noooop committed:
[Misc] Use model_redirect to redirect the model name to a local folder. (#14116)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

zou3519 committed:
Fix incorrect filenames in vllm_compile_cache.py (#15494)

Signed-off-by: <zou3519@gmail.com>
Signed-off-by: youkaichao <youkaichao@gmail.com>
Co-authored-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Doc] update --system for transformers installation in docker doc (#15616)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Model] MiniCPM-V/O supports V1 (#15487)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

h-sugi committed:
[Bugfix] Fix use_cascade_attention handling for Alibi-based models on vllm/v1 (#15211)

Signed-off-by: h-sugi <h.sugi@ieee.org>
Co-authored-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Doc] Link to onboarding tasks (#15629)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Misc] Replace `is_encoder_decoder_inputs` with `split_enc_dec_inputs` (#15620)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

terrytangyuan committed:
[Feature] Add middleware to log API Server responses (#15593)

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Misc] Avoid direct access of global `mm_registry` in `compute_encoder_budget` (#15621)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Use absolute placement for Ask AI button (#15628)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

NickLucche committed:
[Bugfix][TPU][V1] Fix recompilation (#15553)

Signed-off-by: NickLucche <nlucches@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

clnperez committed:
Correct PowerPC to modern IBM Power (#15635)

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

russellb committed:
[CI] Update rules for applying `tpu` label. (#15634)

Signed-off-by: Russell Bryant <rbryant@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

njhill committed:
[V1] AsyncLLM data parallel (#13923)

Signed-off-by: Nick Hill <nhill@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[TPU] Lazy Import (#15656)

Signed-off-by: rshaw@neuralmagic.com <robertgshaw2@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[Quantization][V1]  BitsAndBytes support V1 (#15611)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kebe7jun committed:
[Bugfix] Fix failure to launch in Tensor Parallel TP mode on macOS. (#14948)

Signed-off-by: Kebe <mail@kebe7jun.com>
Signed-off-by: youkaichao <youkaichao@gmail.com>
Co-authored-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wwl2755 committed:
[Doc] Fix dead links in Job Board (#15637)

Signed-off-by: wwl2755 <wangwenlong2755@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[CI][TPU] Temporarily Disable Quant Test on TPU (#15649)

Signed-off-by: rshaw@neuralmagic.com <robertgshaw2@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wrmedford committed:
Revert "Use Cache Hinting for fused_moe kernel (#15511)" (#15645)

Signed-off-by: Wes Medford <wryanmedford@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

CXIAAAAA committed:
[Misc]add coding benchmark for speculative decoding (#15303)

Signed-off-by: CXIAAAAA <cxia0209@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[Quantization][FP8] Adding support for fp8 gemm layer input in fp8 (#14578)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Co-authored-by: Luka Govedič <ProExpertProg@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

JasonZhu1313 committed:
Refactor error handling for multiple exceptions in preprocessing (#15650)

Signed-off-by: JasonZhu1313 <jasonchu13@outlook.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Bugfix] Fix `mm_hashes` forgetting to be passed (#15668)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[V1] Remove legacy input registry (#15673)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[TPU][CI] Fix TPUModelRunner Test (#15667)

Signed-off-by: Robert Shaw <robshaw@redhat.com>
Co-authored-by: Robert Shaw <robshaw@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gaocegege committed:
[Refactor][Frontend] Keep all logic about reasoning into one class (#14428)

Signed-off-by: Ce Gao <cegao@tensorchord.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bigPYJ1151 committed:
[CPU][CI] Improve CPU Dockerfile (#15690)

Signed-off-by: jiang1.li <jiang1.li@intel.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[Bugfix] Fix 'InductorAdaptor object has no attribute 'cache_dir' (#15674)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lizzzcai committed:
[Misc] Fix test_sleep to use query parameters (#14373)

Signed-off-by: Lize Cai <lize.cai@sap.com>
Signed-off-by: youkaichao <youkaichao@gmail.com>
Co-authored-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gaocegege committed:
[Bugfix][Frontend] Eliminate regex based check in reasoning full generator (#14821)

Signed-off-by: Ce Gao <cegao@tensorchord.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Frontend] update priority for --api-key and VLLM_API_KEY (#15588)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
[Docs] Add "Generation quality changed" section to troubleshooting (#15701)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[Model] Adding torch compile annotations to chatglm (#15624)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chaunceyjiang committed:
[Bugfix][v1] xgrammar structured output supports Enum. (#15594)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Bugfix] `embed_is_patch` for Idefics3 (#15696)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

russellb committed:
[V1] Support disable_any_whtespace for guidance backend (#15584)

Signed-off-by: Russell Bryant <rbryant@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[doc] add missing imports (#15699)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kebe7jun committed:
[Bugfix] Fix regex compile display format (#15368)

Signed-off-by: Kebe <mail@kebe7jun.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
Fix cpu offload testing for gptq/awq/ct (#15648)

Signed-off-by: mgoin <mgoin64@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[Minor] Remove TGI launching script  (#15646)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Misc] Remove unused utils and clean up imports (#15708)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ShangmingCai committed:
[Misc] Remove stale func in KVTransferConfig (#14746)

Signed-off-by: Shangming Cai <caishangming@linux.alibaba.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[TPU] [Perf] Improve Memory Usage Estimation (#15671)

Signed-off-by: Robert Shaw <robshaw@redhat.com>
Co-authored-by: Robert Shaw <robshaw@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ProExpertProg committed:
[Bugfix] [torch.compile] Add Dynamo metrics context during compilation (#15639)

Signed-off-by: luka <luka@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexm-redhat committed:
[V1] TPU - Fix the chunked prompt bug (#15713)

Signed-off-by: Alexander Matveev <amatveev@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Misc] cli auto show default value (#15582)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

daniel-salib committed:
implement prometheus fast-api-instrumentor for http service metrics (#15657)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

simpx committed:
[Docs][V1] Optimize diagrams in prefix caching design (#15716)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm][AMD][Build] Update AMD supported arch list (#15632)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

pengyuange committed:
[Model] Support Skywork-R1V (#15397)

Signed-off-by: jiacai.liu <932997367@qq.com>
Co-authored-by: jiacai.liu <932997367@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gaocegege committed:
[Docs] Document v0 engine support in reasoning outputs (#15739)

Signed-off-by: Ce Gao <cegao@tensorchord.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

njhill committed:
[Misc][V1] Misc code streamlining (#15723)

Signed-off-by: Nick Hill <nhill@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

varun-sundar-rabindranath committed:
[Bugfix] LoRA V1: add and fix entrypoints tests (#15715)

Signed-off-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Co-authored-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

russellb committed:
[CI] Speed up V1 structured output tests (#15718)

Signed-off-by: Russell Bryant <rbryant@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

cyyever committed:
Use numba 0.61 for python 3.10+ to support numpy>=2 (#15692)

Signed-off-by: cyy <cyyever@outlook.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jinzhen-lin committed:
[Bugfix] set VLLM_WORKER_MULTIPROC_METHOD=spawn for vllm.entrypoionts.openai.api_server (#15700)

Signed-off-by: Jinzhen Lin <linjinzhen@hotmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

NickLucche committed:
[TPU][V1][Bugfix] Fix w8a8 recompiilation with GSM8K (#15714)

Signed-off-by: NickLucche <nlucches@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yarongmu-google committed:
[Kernel][TPU][ragged-paged-attn] vLLM code change for PR#8896 (#15659)

Signed-off-by: Yarong Mu <ymu@google.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[doc] update doc (#15740)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

tjtanaa committed:
[FEAT] [ROCm] Add AITER int8 scaled gemm kernel (#15433)

Signed-off-by: tjtanaa <tunjian.tan@embeddedllm.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wwl2755 committed:
[V1] [Feature] Collective RPC (#15444)

Signed-off-by: wwl2755 <wangwenlong2755@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ShangmingCai committed:
[Feature][Disaggregated] Support XpYd disaggregated prefill with MooncakeStore (#12957)

Signed-off-by: Shangming Cai <caishangming@linux.alibaba.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ywang96 committed:
[V1] Support interleaved modality items (#15605)

Signed-off-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[V1][Minor] Simplify rejection sampler's parse_output (#15741)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Isotr0py committed:
[Bugfix] Fix Mllama interleaved images input support (#15564)

Signed-off-by: Isotr0py <2037008807@qq.com>
Co-authored-by: Chen Zhang <zhangch99@outlook.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chaunceyjiang committed:
[CI] xgrammar structured output supports Enum. (#15757)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

juliendenize committed:
[Bugfix] Fix Mistral guided generation using xgrammar (#15704)

Signed-off-by: Julien Denize <julien.denize@mistral.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[doc] update conda to usage link in installation (#15761)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

pansicheng committed:
fix test_phi3v (#15321)

Signed-off-by: pansicheng <sicheng.pan.chn@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[V1] Override `mm_counts` for dummy data creation (#15703)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: lint fix a ruff checkout syntax error (#15767)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kylehh committed:
[Bugfix] Added `embed_is_patch` mask for fuyu model (#15731)

Signed-off-by: Kyle Huang <kylhuang@nvidia.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: Comments to English for better dev experience (#15768)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[V1][Scheduler] Avoid calling `_try_schedule_encoder_inputs` for every request (#15778)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lcy4869 committed:
[Misc] update the comments (#15780)

Signed-off-by: chengyang liu <lcy4869@gmail.com>
Co-authored-by: chengyang liu <lcy4869@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

JenZhao committed:
[Benchmark] Update Vision Arena Dataset and HuggingFaceDataset Setup (#15748)

Signed-off-by: Jennifer Zhao <ai.jenniferzhao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

charlifu committed:
[Feature][ROCm]Enable fusion pass for torch.compile on ROCm (#15050)

Signed-off-by: charlifu <charlifu@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Recommend developing with Python 3.12 in developer guide (#15811)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: better install requirement for install in setup.py (#15796)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

youkaichao committed:
[V1] Fully Transparent Implementation of CPU Offloading (#15354)

Signed-off-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Naveassaf committed:
[Model] Update support for NemotronNAS models (#15008)

Signed-off-by: Nave Assaf <nassaf@nvidia.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alex-jw-brooks committed:
[Bugfix] Fix Crashing When Loading Modules With Batchnorm Stats (#15813)

Signed-off-by: Alex-Brooks <Alex.Brooks@ibm.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

noc-turne committed:
[Bugfix] Fix missing return value in load_weights method of adapters.py (#15542)

Signed-off-by: noc-turne <2270929247@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Upgrade `transformers` to `v4.50.3` (#13905)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[Bugfix] Check dimensions of multimodal embeddings in V1 (#15816)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ShangmingCai committed:
[V1][Spec Decode] Remove deprecated spec decode config params (#15466)

Signed-off-by: Shangming Cai <caishangming@linux.alibaba.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: change GB to GiB in logging close #14979 (#15807)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexm-redhat committed:
[V1] TPU CI - Add basic perf regression test (#15414)

Signed-off-by: Alexander Matveev <amatveev@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Fix Transformers backend compatibility check (#15290)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

markmc committed:
[V1][Core] Remove unused speculative config from scheduler (#15818)

Signed-off-by: Mark McLoughlin <markmc@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Move dockerfiles into their own directory (#14549)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ilmarkov committed:
[Distributed] Add custom allreduce support for ROCM (#14125)

Signed-off-by: ilmarkov <imarkov@redhat.com>
Co-authored-by: ilmarkov <imarkov@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Rename fallback model and refactor supported models section (#15829)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kinfey committed:
[Frontend] Add Phi-4-mini function calling support (#14886)

Signed-off-by: Kinfey <kinfeylo@microsoft.com>
Co-authored-by: Cyrus Leung <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yma11 committed:
[Bugfix][Model] fix mllama multi-image (#14883)

Signed-off-by: yan ma <yan.ma@intel.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

haochengxia committed:
[Bugfix] Fix extra comma (#15851)

Signed-off-by: haochengxia <xhc_1007@163.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexwl committed:
[Bugfix]: Fix is_embedding_layer condition in VocabParallelEmbedding  (#15824)

Signed-off-by: alexwl <alexey.a.kiryushin@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

alexm-redhat committed:
[V1] TPU - Fix fused MOE (#15834)

Signed-off-by: Alexander Matveev <amatveev@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lionelvillard committed:
[sleep mode] clear pytorch cache after sleep (#15248)

Signed-off-by: <villard@us.ibm.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm] Use device name in the warning (#15838)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

heheda12345 committed:
[V1] Implement sliding window attention in kv_cache_manager (#14097)

Signed-off-by: Chen Zhang <zhangch99@outlook.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: can not use uv run collect_env close #13888 (#15792)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wayzeng committed:
[Feature] specify model in config.yaml (#15798)

Signed-off-by: weizeng <weizeng@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

varun-sundar-rabindranath committed:
[Misc] Enable V1 LoRA by default (#15320)

Signed-off-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Co-authored-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ShangmingCai committed:
[Misc] Fix speculative config repr string (#15860)

Signed-off-by: Shangming Cai <caishangming@linux.alibaba.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
[Docs] Fix small error in link text (#15868)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Isotr0py committed:
[Bugfix] Fix no video/image profiling edge case for `MultiModalDataParser` (#15828)

Signed-off-by: Isotr0py <2037008807@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ruisearch42 committed:
[Misc] Use envs.VLLM_USE_RAY_COMPILED_DAG_CHANNEL_TYPE (#15831)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>
Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>
Co-authored-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Commit:
setup correct nvcc version with CUDA_HOME (#15725)

Signed-off-by: Yang Chen <yangche@fb.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
[Model] Support Mistral3 in the HF Transformers format (#15505)

Signed-off-by: mgoin <mgoin64@gmail.com>
Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Co-authored-by: DarkLight1337 <tlleungac@connect.ust.hk>
Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Misc] remove unused script (#15746)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Remove `format.sh` as it's been unsupported >70 days (#15884)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

noooop committed:
[New Model]: jinaai/jina-reranker-v2-base-multilingual  (#15876)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Co-authored-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Commit:
[Doc] Quark quantization documentation (#15861)

Signed-off-by: chaow <chaow@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Reinstate `format.sh` and make `pre-commit` installation simpler (#15890)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Isotr0py committed:
[Misc] Allow using OpenCV as video IO fallback (#15055)

Signed-off-by: Isotr0py <2037008807@qq.com>
Co-authored-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm][Build][Bugfix] Bring the base dockerfile in sync with the ROCm fork (#15820)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bnellnm committed:
Add option to use DeepGemm contiguous grouped gemm kernel for fused MoE operations. (#13932)

Signed-off-by: Bill Nell <bnell@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[CI/Build] Clean up LoRA tests (#15867)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

JenZhao committed:
[Model] Aya Vision (#15441)

Signed-off-by: Jennifer Zhao <ai.jenniferzhao@gmail.com>
Signed-off-by: Roger Wang <ywang@roblox.com>
Co-authored-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

cloud11665 committed:
[Model] Add module name prefixes to gemma3 (#15889)

Signed-off-by: Bartholomew Sabat <bartek@recursal.ai>
Co-authored-by: Bartholomew Sabat <bartek@recursal.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ywang96 committed:
[CI] Disable flaky structure decoding test temporarily. (#15892)

Signed-off-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

markmc committed:
[V1][Metrics] Initial speculative decoding metrics (#15151)

Signed-off-by: Mark McLoughlin <markmc@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[V1][Spec Decode] Implement Eagle Proposer [1/N] (#15729)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

simon-mo committed:
[Docs] update usage stats language (#15898)

Signed-off-by: simon-mo <simon.mo@hey.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
[BugFix] make sure socket close (#15875)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ZZBoom committed:
[Model][MiniMaxText01] Support MiniMaxText01 model inference (#13454)

Signed-off-by: qscqesze <475517977@qq.com>
Co-authored-by: qingjun <qingjun@minimaxi.com>
Co-authored-by: qscqesze <475517977@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

simon-mo committed:
[Docs] Add Ollama meetup slides (#15905)

Signed-off-by: simon-mo <simon.mo@hey.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

simon-mo committed:
[Docs] Add Intel as Sponsor (#15913)

Signed-off-by: simon-mo <simon.mo@hey.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ekagra-ranjan committed:
[Spec Decode] Fix input triton kernel for eagle (#15909)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

b8zhong committed:
[V1] Fix: make sure `k_index` is int64 for `apply_top_k_only` (#15907)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gau-nernst committed:
[Bugfix] Fix imports for MoE on CPU (#15841)

Signed-off-by: Thien Tran <gau.nernst@yahoo.com.sg>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[V1][Minor] Enhance SpecDecoding Metrics Log in V1 (#15902)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chun37 committed:
[Doc] Update rocm.inc.md (#15917)

Signed-off-by: chun37 <chun.jb.37@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ywang96 committed:
[V1][Bugfix] Fix typo in MoE TPU checking (#15927)

Signed-off-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Potabk committed:
[Benchmark]Fix error message (#15866)

Signed-off-by: wangli <wangli858794774@gmail.com>
Co-authored-by: Roger Wang <136131678+ywang96@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chaunceyjiang committed:
[Misc] Replace print with logger (#15923)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[CI/Build] Further clean up LoRA tests (#15920)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gau-nernst committed:
[Bugfix] Fix cache block size calculation for CPU MLA (#15848)

Signed-off-by: Thien Tran <gau.nernst@yahoo.com.sg>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

cthi committed:
[Build/CI] Update lm-eval to 0.4.8 (#15912)

Signed-off-by: Chris Thi <chris.c.thi@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

LukasBluebaum committed:
[Kernel] Add more dtype support for GGUF dequantization (#15879)

Signed-off-by: lukas.bluebaum <lukas.bluebaum@aleph-alpha.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

erictang000 committed:
[core] Add tags parameter to wake_up() (#15500)

Signed-off-by: Eric <erictang000@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

russellb committed:
[V1] Fix json_object support with xgrammar (#15488)

Signed-off-by: Russell Bryant <rbryant@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Add minimum version for `huggingface_hub` to enable Xet downloads (#15873)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

b8zhong committed:
[Bugfix][Benchmarks] Ensure `async_request_deepspeed_mii` uses the OpenAI choices key (#15926)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yankay committed:
[CI] Remove duplicate entrypoints-test (#15940)

Signed-off-by: Kay Yan <kay.yan@daocloud.io>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chaunceyjiang committed:
[Bugfix] Fix the issue where the model name is empty string, causing no response with the model name. (#15938)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

markmc committed:
[Metrics] Hide deprecated metrics (#15458)

Signed-off-by: Mark McLoughlin <markmc@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

meffmadd committed:
[Frontend] Implement Tool Calling with `tool_choice='required'` (#13483)

Signed-off-by: Liangfu Chen <liangfc@amazon.com>
Signed-off-by: Matt, Matthias <matthias.matt@tuwien.ac.at>
Co-authored-by: Liangfu Chen <liangfc@amazon.com>
Co-authored-by: mgoin <michael@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bigPYJ1151 committed:
[CPU][Bugfix] Using custom allreduce for CPU backend (#15934)

Signed-off-by: jiang1.li <jiang1.li@intel.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lengrongfu committed:
[Model] use AutoWeightsLoader in model load_weights (#15770)

Signed-off-by: rongfu.leng <rongfu.leng@daocloud.io>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jeejeelee committed:
[Misc] V1 LoRA support CPU offload (#15843)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

npanpaliya committed:
Restricted cmake to be less than version 4 as 4.x breaks the build of… (#15859)

Signed-off-by: Nishidha Panpaliya <nishidha.panpaliya@partner.ibm.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

youkaichao committed:
[misc] instruct pytorch to use nvml-based cuda check (#15951)

Signed-off-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
[V1] Support Mistral3 in V1 (#15950)

Signed-off-by: mgoin <mgoin64@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Fix `huggingface-cli[hf-xet]` -> `huggingface-cli[hf_xet]` (#15969)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hyeygit committed:
[V1][TPU] TPU-optimized top-p implementation (avoids scattering). (#15736)

Signed-off-by: Hyesoo Yang <hyeygit@gmail.com>
Co-authored-by: root <root@t1v-n-822696b7-w-0.us-central2-b.c.tpu-prod-env-large-adhoc.internal>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yaochengji committed:
[TPU] optimize the all-reduce performance (#15903)

Signed-off-by: Chengji Yao <chengjiyao@google.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

NickLucche committed:
[V1][TPU] Do not compile sampling more than needed (#15883)

Signed-off-by: NickLucche <nlucches@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

maleksan85 committed:
[ROCM][KERNEL] Paged attention for V1 (#15720)

Signed-off-by: Aleksandr Malyshev <maleksan@amd.com>
Signed-off-by: root <root@banff-cyxtera-s65-4.amd.com>
Co-authored-by: Aleksandr Malyshev <maleksan@amd.com>
Co-authored-by: root <root@banff-cyxtera-s65-4.amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: better error message for get_config close #13889 (#15943)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

youkaichao committed:
[bugfix] add seed in torchrun_example.py (#15980)

Signed-off-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

maleksan85 committed:
[ROCM][V0] PA kennel selection when no sliding window provided (#15982)

Signed-off-by: Aleksandr Malyshev <maleksan@amd.com>
Co-authored-by: Aleksandr Malyshev <maleksan@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

StevenShi-23 committed:
[Benchmark] Add AIMO Dataset to Benchmark (#15955)

Signed-off-by: Ziji Shi <shi.ziji.sm@gmail.com>
Signed-off-by: StevenShi-23 <shi.ziji.sm@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

youkaichao committed:
[misc] improve error message for "Failed to infer device type" (#15994)

Signed-off-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

wwl2755 committed:
[Bugfix][V1] Fix bug from putting llm_engine.model_executor in a background process (#15367)

Signed-off-by: wwl2755 <wangwenlong2755@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[doc] update contribution link (#15922)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: tiny fix make format.sh excutable (#16015)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kylesayrs committed:
[SupportsQuant] Bert, Blip, Blip2, Bloom (#15573)

Signed-off-by: Kyle Sayers <kylesayrs@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

kylesayrs committed:
[SupportsQuant] Chameleon, Chatglm, Commandr (#15952)

Signed-off-by: Kyle Sayers <kylesayrs@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

liangfu committed:
[Neuron][kernel] Fuse kv cache into a single tensor (#15911)

Signed-off-by: Liangfu Chen <liangfc@amazon.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bnellnm committed:
[Minor] Fused experts refactor (#15914)

Signed-off-by: Bill Nell <bnell@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yarongmu-google committed:
[Misc][Performance] Advance tpu.txt to the most recent nightly torch … (#16024)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Alexei-V-Ivanov-AMD committed:
Re-enable the AMD Testing for the passing tests. (#15586)

Signed-off-by: Alexei V. Ivanov <alexei.ivanov@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

vanbasten23 committed:
[TPU] Support sliding window and logit soft capping in the paged attention kernel for TPU. (#15732)

Signed-off-by: Xiongfei Wei <isaacwxf23@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

robertgshaw2-redhat committed:
[TPU] Switch Test to Non-Sliding Window (#15981)

Signed-off-by: Robert Shaw <robshaw@redhat.com>
Co-authored-by: Robert Shaw <robshaw@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bnellnm committed:
[Bugfix] Fix function names in test_block_fp8.py (#16033)

Signed-off-by: Bill Nell <bnell@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

huydhn committed:
[ROCm] Tweak the benchmark script to run on ROCm (#14252)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Misc] improve gguf check (#15974)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yaochengji committed:
[TPU][V1] Remove ragged attention kernel parameter hard coding (#16041)

Signed-off-by: Chengji Yao <chengjiyao@google.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
doc: add info for macos clang errors (#16049)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

markmc committed:
[V1][Spec Decode] Avoid logging useless nan metrics (#16023)

Signed-off-by: Mark McLoughlin <markmc@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jonghyunchoe committed:
[Model] use AutoWeightsLoader for baichuan, gpt-neox, mpt (#15939)

Signed-off-by: Jonghyun Choe <andy.choe729@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

zhenwei-intel committed:
[Hardware][Gaudi][BugFix] fix arguments of hpu fused moe (#15945)

Signed-off-by: zhenwei <zhenweiliu@habana.ai>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Isotr0py committed:
[Bugfix][kernels] Fix half2float conversion in gguf kernels (#15995)

Signed-off-by: Isotr0py <2037008807@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

StevenShi-23 committed:
[Benchmark][Doc] Update throughput benchmark and README (#15998)

Signed-off-by: StevenShi-23 <shi.ziji.sm@gmail.com>
Signed-off-by: Roger Wang <ywang@roblox.com>
Co-authored-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bigPYJ1151 committed:
[CPU] Change default block_size for CPU backend (#16002)

Signed-off-by: jiang1.li <jiang1.li@intel.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ilmarkov committed:
[Distributed] [ROCM] Fix custom allreduce enable checks (#16010)

Signed-off-by: ilmarkov <imarkov@redhat.com>
Co-authored-by: ilmarkov <imarkov@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm][Bugfix] Use platform specific FP8 dtype (#15717)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

gshtras committed:
[ROCm][Bugfix] Bring back fallback to eager mode removed in #14917, but for ROCm only (#15413)

Signed-off-by: Gregory Shtrasberg <Gregory.Shtrasberg@amd.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
[Bugfix] Fix default behavior/fallback for pp in v1 (#16057)

Signed-off-by: mgoin <mgoin64@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

khluu committed:
[CI] Reorganize .buildkite directory (#16001)

Signed-off-by: kevin <kevin@anyscale.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

njhill committed:
[V1] DP scale-out (1/N): Use zmq ROUTER/DEALER sockets for input queue (#15906)

Signed-off-by: Nick Hill <nhill@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

DarkLight1337 committed:
[V1] Scatter and gather placeholders in the model runner (#15712)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>
Signed-off-by: mgoin <mgoin64@gmail.com>
Signed-off-by: Roger Wang <ywang@roblox.com>
Co-authored-by: mgoin <mgoin64@gmail.com>
Co-authored-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ywang96 committed:
Revert "[V1] Scatter and gather placeholders in the model runner" (#16075)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bnellnm committed:
[Kernel][Minor] Re-fuse triton moe weight application (#16071)

Signed-off-by: Bill Nell <bnell@redhat.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

mgoin committed:
[Bugfix][TPU] Fix V1 TPU worker for sliding window (#16059)

Signed-off-by: Michael Goin <mgoin64@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

WoosukKwon committed:
[V1][Spec Decode] Update N-gram Proposer Interface (#15750)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

tristanleclercq committed:
[Misc] Auto detect bitsandbytes pre-quantized models (#16027)

Signed-off-by: Tristan Leclercq <tristanleclercq@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

khluu committed:
[CI] Fix benchmark script level (#16089)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
fix: support clang17 for macos and fix the real libomp (#16086)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[doc] fix 404 (#16082)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

yihong0618 committed:
Revert "doc: add info for macos clang errors (#16049)" (#16091)

Signed-off-by: yihong0618 <zouzou0208@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hmellor committed:
Fix some capitalisations in generated examples doc titles (#16094)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Misc] format output for encoder_decoder.py (#16095)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

chaunceyjiang committed:
[Misc] Remove redundant code (#16098)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jinzhen-lin committed:
[Bugfix] fix use_atomic_add support of marlin kernel when using v1 engine (#15946)

Signed-off-by: Jinzhen Lin <linjinzhen@hotmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

jonghyunchoe committed:
[Model] use AutoWeightsLoader for phi, gemma, deepseek (#16088)

Signed-off-by: Jonghyun Choe <andy.choe729@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

luccafong committed:
[Model] fix model testing for TeleChat2ForCausalLM and V0 llama4 (#16112)

Signed-off-by: Lu Fang <fanglu@fb.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

hyeygit committed:
[Benchmark] Add sampling parameters to benchmark_serving. (#16022)

Signed-off-by: Hyesoo Yang <hyeygit@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

bjj committed:
[Frontend] Fix typo in tool chat templates for llama3.2 and toolace (#14501)

Signed-off-by: Ben Jackson <ben@ben.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

ywang96 committed:
[CI][V1] Fix passing `tokenizer` as kwarg to `validate_guidance_grammar` (#16117)

Signed-off-by: Roger Wang <ywang@roblox.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

reidliu41 committed:
[Misc] refactor example eagle (#16100)

Signed-off-by: reidliu41 <reid201711@gmail.com>
Co-authored-by: reidliu41 <reid201711@gmail.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

paolovic commented:
Hi @lulmer ,
I fixed the causes for the failing tests.
Do you want to incorporate them into your PR or should I create a separate one?

psschwei committed:
[Doc][Bugfix] Add missing EOF in k8s deploy doc (#16025)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

Isotr0py committed:
[Misc] Improve model redirect to accept json dictionary (#16119)

Signed-off-by: Isotr0py <2037008807@qq.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lengrongfu committed:
[Model] use AutoWeightsLoader for stablelm,starcoder2,zamba2 (#16103)

Signed-off-by: rongfu.leng <rongfu.leng@daocloud.io>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

varun-sundar-rabindranath committed:
[Bugfix] LoRA : Fix the order in which the kernels process LoRAs  (#16040)

Signed-off-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Co-authored-by: Varun Sundar Rabindranath <varun@neuralmagic.com>
Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer commented:
> Hi @lulmer , I fixed the causes for the failing tests here: https://github.com/paolovic/vllm/tree/feature/llama_tool_calling . Do you want to incorporate them into your PR or should I create a separate one?

Thank you for your indications @paolovic, I didn't noticed I needed to address linting issues, I updated my branch accordingly as it was simple. I rebased with the current branch as well.

lulmer committed:
fix: fixing typing deprecations issues

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer committed:
fix: fixing typing deprecations issues

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer committed:
fix: fixing typing deprecations issues (3)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer committed:
fix: fixing typing deprecations issues (4)

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer committed:
fix: running pre commits

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

paolovic commented:
Hi @lulmer ,

no problem, please don't forget to sign-off your commits with `-s`, e.g.
`git commit -m "this is an example commit" -s`, otherwise DCO will fail.

As it did now.
I think you can fix it with the instructions given [here](https://github.com/vllm-project/vllm/pull/12752/checks?check_run_id=40058538965) under "Rebase the branch"

lulmer committed:
fix: solving merge conflict due to rebase to comply with DCO

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer committed:
fix: adding pre commit for examples/offline_inference/mistral-small.py

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

lulmer commented:
> Hi @lulmer ,
> 
> no problem, please don't forget to sign-off your commits with `-s`, e.g. `git commit -m "this is an example commit" -s`, otherwise DCO will fail.
> 
> As it did now. I think you can fix it with the instructions given [here](https://github.com/vllm-project/vllm/pull/12752/checks?check_run_id=40058538965) under "Rebase the branch"

@paolovic Thank you, I followed the procedure you've pinpointed, but a lot of unrelated files/labels have been added in this PR, I had to solve a merge conflict (systematically chose the latest updates from main branch) and add back pre commits for 
`examples/offline_inference/mistral-small.py`.
Also I've noticed `core/v1/kv_cache_manager.py` is now part of the modified files, I hope it is expected and won't be a problem for the review. Otherwise let me know what I should do next.

paolovic commented:
Hi @lulmer ,
is it possible that you are, unintentionally, try to push an outdated version of `v1/core/kv_cache_manager.py` to `vllm:main`?
If so, please make sure to sync your fork like so:

lulmer commented:
> Hi @lulmer , is it possible that you are, unintentionally, try to push an outdated version of `v1/core/kv_cache_manager.py` to `vllm:main`? If so, please make sure to sync your fork like so:
> 
> 1. Go to your fork
> 2. Sync fork (on the top right corner)
> 3. Update Branch (Green Button)

I ensured that I synced my fork when I did the rebase (If you see the history, I got told by Mergebot), on [this commit](https://github.com/vllm-project/vllm/pull/12752/commits/3dc264160075a93d15bdb2527c39ffa504f90aed)

When I go to my fork on the github UI, the right tab sync fork says my fork is up to date with the latest main branch, see below.
![image](https://github.com/user-attachments/assets/1a987fb7-6a68-4a79-be03-b0185e35ef3e)

lulmer commented:
@paolovic is there any additional steps I should do now ?

paolovic commented:
@lulmer tbh: i would close this PR and create a new, clean one where `v1/core/kv_cache_manager.py` is not listed as a changed file.

furthermore, fix the failing checks. without them it won't be merged.

FYI: I cannot and won't do the code review.

lulmer committed:
Merge branch 'main' into llama_usr_defined_toolcall_template

lulmer committed:
fix: removing the accidental modification of the kv_cache_manager

Signed-off-by: Louis Ulmer <ulmerlouis@gmail.com>

paolovic commented:
@lulmer nice, super cool!

By the way could you provide an example for me how to use it?
